sourced/ml/algorithms/id_splitter/features.py

```diff
@@ -0,0 +1,108 @@
+import logging
+import string
+import tarfile
+import warnings
+
+import numpy as np
+try:
+    from keras.preprocessing.sequence import pad_sequences
+except ImportError as e:
+    warnings.warn("Tensorflow or/and Keras are not installed, dependent functionality is "
+                  "unavailable.")
+
+
+MAXLEN = 40  # max length of sequence
+PADDING = "post"  # add padding values after input
+
+# CSV default parameters
+TOKEN_COL = 3
+TOKEN_SPLIT_COL = 4
+
+
+def read_identifiers(csv_loc, use_header=True, mode="r", maxlen=MAXLEN, token_col=TOKEN_COL,
+                     token_split_col=TOKEN_SPLIT_COL, shuffle=True):
+    """
+    Read and filter too long identifiers from CSV file.
+    :param csv_loc: location of CSV file
+    :param use_header: use header as normal line (True) or treat as header line with column names
+    :param mode: mode to read tarfile
+    :param maxlen: maximum length of raw identifier. If it's longer than skip it
+    :param token_col: column in CSV file for raw token
+    :param token_split_col: column in CSV file for splitted token
+    :param shuffle: shuffle or not list of identifiers
+    :return: list of splitted tokens
+    """
+    log = logging.getLogger("id-splitter-prep")
+    log.info("Reading data from CSV...")
+    identifiers = []
+    with tarfile.open(csv_loc, mode=mode, encoding="utf-8") as f:
+        assert len(f.members) == 1, "Expect one archived file"
+        content = f.extractfile(f.members[0])
+        if not use_header:
+            content.readline()
+        for line in content:
+            parts = line.decode("utf-8").strip().split(",")
+            if len(parts[token_col]) <= maxlen:
```

Comment:
Instead of setting the max length why not add it as an argument which defaults to 40 ?